semgrep-core/src/parsing/tree_sitter/Parse_go_tree_sitter.ml

```diff
@@ -51,8 +51,20 @@ let rev = false
    to another type of tree.
 *)
 
-(* TODO: need remove enclosing `` *)
-let raw_string_literal env tok = str env tok
+(* TODO: Update grammar so that the leading and trailing backticks are tokenized
+ * separately, the way interpreted string literals are:
+ * https://github.com/tree-sitter/tree-sitter-go/blob/0fa917a7022d1cd2e9b779a6a8fc5dc7fad69c75/grammar.js#L839-L843
+ * *)
+let raw_string_literal env tok =
+  let loc, s = tok in
+  (* Remove leading and trailing backticks. The grammar guarantees that raw
+   * string literals will always have leading and trailing backticks, so this
+   * String.sub call should be safe. Let's check just to be sure. *)
+  if String.length s < 2 || s.[0] <> '`' || s.[String.length s - 1] <> '`' then
+    failwith "Found unexpected raw string literal without delimiters";
+  let s = String.sub s 1 (String.length s - 2) in
+  let tok = (loc, s) in
+  str env tok
```

Comment:
We could simply add `EncodedString` here if we wanted to. The difference is that the way this is encoded, a pattern:
```
`foo`
```
will match a code expression `"foo"`, they are encoded as equivalent. For Python, Semgrep can match pattern `"foo"` against `r"foo"` via constant folding, but not the other way around.

I am not arguing in favor or against any approach, but I think we should be consistent.

Reply:
That makes sense, and it sounds like something that should be done. I'd prefer to leave it for another PR, though, if that's okay with you.

Reply:
I think the change is so small that it's better to do it in this PR, something like should work:

```ocaml
Call(IdSpecial(EncodedString "r", tok), (tok,[str env tok],tok))
```

Otherwise soon enough somebody will complain about why this behaves "this way" in Python and "this other way" in Go.

Reply:
Well, it may not be that small, because `string_literal` is not supposed to return an expression.

Reply:
I've looked into this and if I understand correctly, this would require a change to the pfff AST for Go. Right now, when we translate from the tree-sitter CST to the pfff AST, all string literals are represented by the `String` variant of the `literal` type defined in `semgrep-core/src/pfff/lang_go/parsing/ast_go.ml`. We'd need another variant there to differentiate between kinds of string literals, so that we could then choose `EncodedString` rather than `String` in `go_to_generic.ml` when we encounter a raw string literal.

Unfortunately, I don't currently have the time to make such a change. I hope that you can agree that as-is, this PR is still a step forward and is worth merging, but I understand if you'd rather wait until someone is able to make your requested changes.

Reply:
Yeah that sounds right! I think @aryx wants to merge anyways. No big deal for me. I just thought it's better to fix it consistently with current behavior for other languages, unless this is blocking some customer (which is not AFAIK). Good work anyways!